pkg/kubelet/server.go

```diff
@@ -63,6 +67,11 @@ type TLSOptions struct {
 	KeyFile  string
 }
 
+// RequestAttributeGetter is a function that extracts authorizer.Attributes for a given user from a restful.Request
+type AuthorizerAttributeGetter interface {
+	GetAuthorizerAttributes(user.Info, *restful.Request) (authorizer.Attributes, error)
```

Comment:
The `*restful.Request` param seems a little specific for this interface -- how about an interface like so:

```go
type RequestInterface interface {
  SetAttribute(name string, value interface{})
  Attribute(name string) interface{}
}
```

Maybe that's overkill here, it just smells funny to have to pull in go-restful to use the `AuthorizerAttributeGetter` interface.